### PR TITLE
ESBJAVA-4287

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttAsyncCallback.java [hunk 1]
```diff
@@ -129,7 +131,6 @@ public void messageArrived(String topic, MqttMessage mqttMessage) throws MqttExc
     }
 
     public void deliveryComplete(IMqttDeliveryToken iMqttDeliveryToken) {
-        log.info("message delivered .. : " + iMqttDeliveryToken.toString());
     }
```
Comment (jagathsisira) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttAsyncCallback.java]: Empty method can be removed

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttConnectionListener.java [hunk 2]
```diff
@@ -37,28 +36,33 @@ public MqttConnectionListener(MqttConnectionConsumer mqttConnectionConsumer) {
...
+                }
 
-            if (execute) {
-                mqttConnectionConsumer.getMqttAsyncClient()
-                        .connect(mqttConnectionConsumer.getConnectOptions(), this);
             }
-        } catch (MqttException ex) {
-            log.error("Error while trying to subscribe to the remote", ex);
-        } catch (InterruptedException ex) {
+        }  catch (InterruptedException ex) {
```
Comment (isururanawaka) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttConnectionListener.java]: I think we need to handle Interrupted Exception properly. ex :-  Thread.currentThread().interrupt() 
But need to focus weather this is necessary.

Comment (erandacr) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttConnectionListener.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttAsyncCallback.java (hunk 1 above)
Comment (erandacr) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttAsyncCallback.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

**Pull request merged**